example/msw.ts

```diff
@@ -1,10 +1,56 @@
 import { setupWorker } from 'msw/browser';
-import { getMswHandlers } from '../src/FakeRest';
+import { MswServer, withDelay } from '../src';
 import { data } from './data';
+import { dataProvider as defaultDataProvider } from './dataProvider';
 
-export const worker = setupWorker(
-    ...getMswHandlers({
-        baseUrl: 'http://localhost:3000',
-        data,
-    }),
-);
+const restServer = new MswServer({
+    baseUrl: 'http://localhost:3000',
+    data,
+});
+
+restServer.addMiddleware(withDelay(300));
+restServer.addMiddleware(async (request, context, next) => {
+    if (!request.headers?.get('Authorization')) {
+        return {
+            status: 401,
+            headers: {},
+        };
+    }
+    return next(request, context);
+});
+
+restServer.addMiddleware(async (request, context, next) => {
+    if (context.collection === 'books' && request.method === 'POST') {
+        if (
+            restServer.collections[context.collection].getCount({
+                filter: {
+                    title: context.requestBody?.title,
+                },
+            }) > 0
+        ) {
+            throw new Response(
+                JSON.stringify({
+                    errors: {
+                        title: 'An article with this title already exists. The title must be unique.',
+                    },
+                }),
+                {
+                    status: 400,
+                    statusText: 'Title is required',
+                },
+            );
```

Comment:
I still use a `throw Response` here instead of returning an object to test that this still works